### ISSUE TITLE
Self-Supervised GANs via Auxiliary Rotation Loss

### ISSUE THREAD

**Issue by KeisukeShimokawa** [KeisukeShimokawa/papers-challenge#124]
### 論文へのリンク

[[arXiv:1811.11212] Self-Supervised GANs via Auxiliary Rotation Loss](https://arxiv.org/abs/1811.11212)

### 著者・所属機関

Ting Chen, Xiaohua Zhai, Marvin Ritter, Mario Lucic, Neil Houlsby

- University of California, Los Angeles
- Google Brain

### 投稿日時（YYYY-MM-DD）

2018-11-27

**Comment by KeisukeShimokawa** [comment]
[https://github.com/vandit15/Self-Supervised-Gans-Pytorch](https://github.com/vandit15/Self-Supervised-Gans-Pytorch)

**Comment by KeisukeShimokawa** [comment]
まずはDataLoaderから取り出した画像を、90度・180度・270度回転させてデータを増強する。

```python
for i, data in enumerate(dataloader):
    # get real/fake image data
    real = data[0]
    fake = netG.sample(batch_size)

    # rotate real image, then concatenate them
    real_90   = real.transpose(3, 2)
    real_180 = real.flip(2, 3)
    real_270 = real.transpose(2, 3).flip(2, 3)
    real = torch.cat((real, real_90, real_180, real_270), dim=0)

    # rotate fake image, then concatenate them
    fake_90   = fake.transpose(3, 2)
    fake_180 = fake.flip(2, 3)
    fake_270 = fake.transpose(2, 3).flip(2, 3)
    fake = torch.cat((fake, fake_90, fake_180, fake_270), dim=0)
```

**Comment by KeisukeShimokawa** [comment]
提案手法の元

- [26] Spyros Gidaris, Praveer Singh, and Nikos Komodakis. Unsupervised representation learning by predicting image rotations. In International Conference on Learning Representations
(ICLR), 2018.
